### PR TITLE
Fixed issue [security] #18355 user can grant more permissions to any …

### PR DESCRIPTION
**Pull request by Shnoulle**
Fixed issue [security] #18355 user can grant more permissions to any user than he has himself
Dev: Update only current user permission
Dev: check to disallow update of himself
Dev: fix the HTML error return (broken before)

### CODE REVIEW

File: application/controllers/UserManagementController.php [hunk 1]
```diff
@@ -1394,40 +1413,90 @@ protected function getRandomString(): string
...
             ];
         }
         return $results;
     }
 
+    /**
+     * Get user permission (CRUD) as array
+     * @return array
+     */
+    private function getCurrentUserPermission(): array
```
Comment (olleharstedt) [inline application/controllers/UserManagementController.php:1462]: Should be in service class?

Comment (Shnoulle) [reply application/controllers/UserManagementController.php:1462]: No, we have it here : https://github.com/LimeSurvey/LimeSurvey/blob/9510ce9580e1f85cc3b1b980e5474413789c78b4/application/controllers/UserManagementController.php#L412

I have to apply it here too : same function twice. I make the commit.

About Service class : unsure it must be the same way for Object ? I check at same time.
(but prefer to have it merged monday)

Comment (Shnoulle) [reply application/controllers/UserManagementController.php]: I move to a single funtcion for both part.

About in Service class : yes we have it , but need a complet GUI rewirite.

We use same Gui (near) for Survey and for SurveyGroup (and hope for user, Template etc …) but here : it's a security fix not a rewrite.

Comment (Shnoulle) [reply application/controllers/UserManagementController.php]: I Survey we use disable : https://github.com/LimeSurvey/LimeSurvey/blob/9510ce9580e1f85cc3b1b980e5474413789c78b4/application/models/services/PermissionManager.php#L77 
Not hidden

**Comment by olleharstedt** [comment]
Is this the smallest possible change to fix the issue?

**Comment by Shnoulle** [comment]
Then: have issue when save etc …

**Comment by olleharstedt** [comment]
> Then: have issue when save etc …

Yes, but when the commit is small, it's easier to review without getting anxious about possible regressions... :neutral_face:

**Pull request closed without merging**